src/controllers/scripting/controllerscriptenginebase.h

```diff
@@ -47,6 +47,7 @@ class ControllerScriptEngineBase : public QObject {
     std::shared_ptr<QJSEngine> m_pJSEngine;
 
     Controller* m_pController;
+    const QLoggingCategory& m_logger;
```

Comment:
Once again, storing references is unsafe without any control of their lifetimes.